lib/ansible/modules/network/meraki/meraki_rest.py

```diff
@@ -0,0 +1,238 @@
+                       method=dict(type='str', choices=['delete', 'get', 'post', 'put'], required=True),
+                       path=dict(type='path', required=True),
+                       timeout=dict(type='int', default=30),
+                       use_proxy=dict(type='bool', default=False),
+                       use_ssl=dict(type='bool', default=True),
+                       validate_certs=dict(type='bool', default=True),
+                       content=dict(type='raw'),
+                       src=dict(type='path'),
+                       output_level=dict(type='str', choices=['normal', 'debug'])
+                       )
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # change is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=False,
+        mutually_exclusive=[['content', 'src']],
+    )
+    module.params['follow_redirects'] = 'urllib2'
+
+    try:
+        module.params['auth_key'] = os.environ['MERAKI_KEY']
+    except KeyError:
+        pass
+
+    if module.params['auth_key'] is None:
+        module.fail_json(msg='Meraki Dashboard API key not set')
+
+    path = module.params['path']
+    payload = None
+
+    if module.params['content']:
+        payload = module.params['content']
+        payload = json.dumps(payload)
+    elif module.params['src']:
+        src = module.params['src']
+        file_exists = False
+        if os.path.isfile(src):
+            file_exists = True
+            with open(src, 'r') as config_object:
+                payload = config_object.read()
+        else:
+            module.fail_json(msg="File does not exist")
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    if module.check_mode:
+        return result
+
+    # manipulate or modify the state as needed (this is going to be the
+    # part where your module will do what it needs to do)
+    protocol = 'https'
+    if module.params['use_ssl'] is not None and module.params['use_ssl'] is False:
+        protocol = 'http'
+
+    url = '{0}://{1}/{2}'.format(protocol, module.params['host'], module.params['path'].lstrip('/'))
+    headers = {'Content-Type': 'application/json',
+               'X-Cisco-Meraki-API-Key': module.params['auth_key'],
+               }
+
+    if module.params['output_level'] == 'debug':
```

Comment:
So, I think on failure this output would be useful as well. So maybe it's better to create a new dictionary `debug_result` which you can then provide to fail_json calls related to fetch_url failing, and you can merge it with result before exit_json.

```python
debug_result = dict(
    url=url,
    method=module.params['method'].upper(),
    headers=headers,
    payload=payload,
)
```

Then add the response and status:
```python
debug_result['status'] = info['status']
debug_result['response'] = resp.read()
```

Then call fail_json using it:
```python
    module.fail_json(msg=str(e), **debug_result)
```

And exit_json:
```python
    if module.params['output_level'] == 'debug':
        result.update(debug_result)

    module.exit_json(**result)
```